### PR TITLE
Do not show modal for trusted repositories

### PR DESCRIPTION
**Pull request by akurinnoy**
### What does this PR do?

This PR fixes
- the behavior of the Untrusted Repository modal, so that it doesn't reappear for already trusted repositories (see "Bug # 1);
- when switching to either the Logs or Events tab during the workspace creation phase, and staying there for a while, then the warning about the limit of running workspaces appears (see "Bug # 2"). The workaround for this bug is to disable the Logs and Events tab until the workspace is fully created;
- do not show the Untrusted Repository modal when starting an existing workspaces;
- do not show the Untrusted Repository modal when creating a workspace from the sample project on the Getting Started page.

### Screenshot/screencast of this PR

<details>
<Summary>Bug # 1</Summary>

https://github.com/user-attachments/assets/0f7653b6-a6cd-4edc-b043-e53c5e1faeca
</details>

<details>
<summary>Bug # 2</summary>

https://github.com/user-attachments/assets/f6b07753-a1ed-4af4-b724-d9fb70722a16
</details>

<details>
<summary>With fixes</summary>

https://github.com/user-attachments/assets/e4662b3f-1e86-42f1-994e-c8d20cf2f988
</details>

### What issues does this PR fix or reference?

fixes https://github.com/eclipse-che/che/issues/23097
fixes https://github.com/eclipse-che/che/issues/23107
fixes https://github.com/eclipse-che/che/issues/23106
fixes [CRW-7139](https://issues.redhat.com//browse/CRW-7139)

### Is it tested? How?

See the screencasts.

### CODE REVIEW

File: build/dockerfiles/skaffold.Dockerfile [hunk 1]
```diff
@@ -10,6 +10,8 @@
 
 FROM docker.io/node:18.19.1-alpine3.19
 
+LABEL quay.expires-after=1w
```
Comment (ibuziuk) [inline build/dockerfiles/skaffold.Dockerfile:13]: was not aware of this feature - https://docs.redhat.com/ja/documentation/red_hat_quay/3.3/html/use_red_hat_quay/tag-expiration 
why do we need it?

**Comment by ibuziuk** [comment]
great follow up PR, should be backported to 7.90.x for 3.16 cc: @dmytro-ndp @artaleks9

File: build/dockerfiles/skaffold.Dockerfile (hunk 1 above)
Comment (akurinnoy) [reply build/dockerfiles/skaffold.Dockerfile:13]: we use this docker file to build images for testing purposes, so we don't need them to be in the registry forever.

**Comment by dmytro-ndp** [comment]
@akurinnoy : thanks for the fixup!

Validation results on Dev Spaces 3.16.0.RC + quay.io/eclipse/che-dashboard:pr-1172
> 1) Do not show the Untrusted Repository modal when starting an existing workspaces
worked as expected

[test_start_existed_workspace.webm](https://github.com/user-attachments/assets/966e665d-9dbb-481e-bb92-877774270d2d)

**Comment by dmytro-ndp** [comment]
@akurinnoy : I have retested new image from this PR: quay.io/eclipse/che-dashboard@sha256:6192db43235b5f1b729e5e8ea5c72d5793d3a527abdf907e488d4ecfc1d9365a

In the updated User Dashboard the Untrusted Author popup didn't appear again when create new workspace from the same factory URL.
Well done!

I still need help with verification of the next fix:
> [4) do not show the Untrusted Repository modal when creating a workspace from the sample project on the Getting Started page](https://issues.redhat.com//browse/CRW-7139)

Provided Dev Spaces configuration didn't work.

**Review by dmytro-ndp — APPROVED** [review]
@akurinnoy : I have managed to test it with Dev Spaces 3.16.0.RC using quay.io/abazko/che-dashboard:CRW-7139 which is quay.io/eclipse/che-dashboard:pr-1172 adapted to run in Dev Spaces by @tolusha .

No issues mentioned in the description of PR have been reproduced:

[che-dashboard_pull_1172_with_devspaces.webm](https://github.com/user-attachments/assets/b85f2acb-d167-4d61-be87-cccbdf20e89a)

Well done!

**Comment by akurinnoy** [comment]
@dmytro-ndp, Thank you!

**Pull request merged**